web/src/js/ducks/modes/utils.ts

```diff
@@ -0,0 +1,81 @@
+import { getMode as getRegularModeConfig } from "./regular";
+import { getMode as getLocalModeConfig } from "./local";
+import { fetchApi } from "../../utils";
+
+export interface ModeState {
+    active: boolean;
+    name: string;
+    listen_port?: number;
+    listen_host?: string;
+    error?: string;
+}
+
+export const updateMode = () => {
+    return async (_, getState) => {
+        try {
+            const modes = getState().modes;
+
+            const activeModes: string[] = [
+                ...getRegularModeConfig(modes),
+                ...getLocalModeConfig(modes),
+                //add new modes here
+            ];
+            const response = await fetchApi.put("/options", {
+                mode: activeModes,
+            });
+            if (response.status === 200) {
+                return { success: true };
+            } else {
+                const errorText = await response.text();
+                return { success: false, error: errorText };
+            }
+        } catch (error) {
+            //TODO: handle error
+        }
+    };
+};
+
+export const addListenAddr = (mode: ModeState) => {
+    let stringMode = mode.name;
+    if (mode.active) {
```

Comment:
Should `addListenAddr` really peek at `.active`? I wouldn't expect that at all from the name. We should move that out of here and do it manually in each relevant `getMode`.

Reply:
mhh okay, but what's the point to move it outside and write it in each `getMode`? The check is always the same and we would repeat the code

Reply:
We can either go with a separate `skipInactive()` helper func, or we come up with a good function name that conveys this method does both. Maybe something like `includeModeState`? We can then do this:

```javascript
    return includeModeState("regular", modes.regular);
```